src/addrman.cpp

```diff
@@ -459,11 +459,14 @@ CAddrInfo CAddrMan::Select_(bool newOnly)
     }
 }
 
-#ifdef DEBUG_ADDRMAN
 int CAddrMan::Check_()
 {
     AssertLockHeld(cs);
 
+    // Run consistency checks 1 in m_consistency_check_ratio times if enabled
+    if (m_consistency_check_ratio == 0) return 0;
+    if (insecure_rand.randrange(m_consistency_check_ratio) >= 1) return 0;
```

Comment:
This is still true, right?

> This has the unfortunate effect that the checks are run on the same operations for every run of the unit tests. Having them picked at random means that eventually they're run at every operation in the unit tests.

From https://github.com/bitcoin/bitcoin/pull/20233#discussion_r672913835

Reply:
Yes, I think this is true, but I'm not sure how big a problem it is. Before this PR we weren't running the consistency tests at all in the unit tests.

Reply:
Test